app/Concern/HasSlug.php

```diff
@@ -0,0 +1,27 @@
+<?php
+namespace App\Concern;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+/**
+ * HasSlug
+ */
+trait HasSlug
+{
+
+    /**
+     * @param null|string $slug
+     */
+    public function setSlugAttribute(?string $slug): void
+    {
+        $attributeName = 'slug';
+        /** @var $this Model */
+        if ($this->name && is_null($slug)) {
```

Comment:
Utilise le ternaire
` $this->attributes[$attributeName]=($this->name && is_null($slug))?Str::slug($this->name):$slug;`

Reply:
Pour cette partie c'est bon, j'ai rajouté les TU pour les slugs.